From 197a876368a98a3d8fff0244cb31a9c224580f09 Mon Sep 17 00:00:00 2001
From: kboeluek <kboeluek@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 16 Jan 2014 15:56:20 +0000
Subject: [PATCH 0514/1076] MANTIS 0025026 Mantis 0025026: Report process
 counters to logging service LogManager_impl.java logs counter informations.

platform.idl
V1.3


git-svn-id: svn://localhost/SelfServiceCommon/trunk@639 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/platform.idl            | 22 +++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 881158ae..51544a89 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -23,12 +23,13 @@
 
 /** This module provides interfaces for the platform internal communication between
  *  the platform management and the device components.
- *  @version 1.2
  *  @author Carsten Metzler - Materna Information & Communications
  */
 
 module PlatformManagement
 {
+    const string interface_version = "1.3";
+
     /** Data type for text lists. */
 
     typedef sequence<string> list;
@@ -209,14 +210,27 @@ module PlatformManagement
                                 //   LC_BIN* not used in LoggingComponentData
       string moduleName;	// needed for GAC_MODULENAME
     };
+	
+	struct LoggingCounterData
+	{
+	  string realComponentName;
+	  LoggingClass unitClass;
+	  long bin;                 //bin number
+	  long mediaStatusTypeValue;//int value of mediaStatusType
+	  long couponsproduced;
+	  long numberOfDocs;
+	  long paperType;           //org.iata.cuss.Characteristics.MediaOutputPackage;
+	  string paperClass;        //com.materna.massai.DeviceComponents.characteristics.types
+	};
     
     interface PlatformIF
     {
 
 	typedef sequence<LoggingComponentData> LoggingComponentDataList;   
-    	void initLogging(in string reladComponentName, in LoggingComponentDataList init);
-
-
+    	void initLogging(in string realComponentName, in LoggingComponentDataList init);
+        void logCounter(in LoggingCounterData lcd);
+        void logCounterFailure(in string realComponentName, in LoggingClass unitClass, in long numberOfFailed);
+		
         /** Checks the status of the token passed by the application.
          *  @param token - The application token
          *  @return the token status of the application token.
-- 
2.41.0.windows.1

